File: external/java/src/main/java/ch/epfl/dedis/lib/darc/Identity.java
package ch.epfl.dedis.lib.darc;

import ch.epfl.dedis.lib.proto.DarcProto;

public interface Identity {
    /**
     * Returns true if the verification of signature on the sha-256 of msg is
     * successful or false if not.
     * @param msg the message
     * @param signature the signature
     * @return true if the signature is correct
     */
    boolean verify(byte[] msg, byte[] signature);

    /**
     * Creates a protobuf-representation of the implementation. The protobuf
     * representation has to hold all necessary fields to represent any of the
     * identity implementations.
     * @return the protobuf-representation of the Identity
     */
    DarcProto.Identity toProto();

    boolean equals(Object other);

    String typeString();

    String toString();

    /**
     * Get the public identifier in bytes. The returned value is not a serialisation of the identity.
     */
    byte[] getPublicBytes();
}

File: external/js/cothority/src/darc/identity.ts
import IdentityWrapper from "./identity-wrapper";

/**
 * Identitiy is an abstract class for all the Darcs's identities
 */
export default interface Identity {
  /**
   * Returns true if the verification of signature on the sha-256 of msg is
   * successful or false if not.
   * @param msg       the message to verify
   * @param signature the signature to verify
   * @returns true when the signature matches the message, false otherwise
   */
  verify(msg: Buffer, signature: Buffer): boolean;

  /**
   * Get the type of the identity
   * @returns the type of the identity as a string
   */
  typeString(): string;

  /**
   * Get the wrapper used to encode the identity
   * @returns the wrapper
   */
  toWrapper(): IdentityWrapper;

  /**
   * Get the byte array representation of the public key of the identity
   * @returns the public key as buffer
   */
  toBytes(): Buffer;

  /**
   * Get the string representation of the identity
   * @return a string representation
   */
  toString(): string;
}
